Extension/src/LanguageServer/client.ts

```diff
@@ -2204,6 +2214,11 @@ export class DefaultClient implements Client {
         return this.languageClient.sendRequest(IncludesRequest, params);
     }
 
+    public async getChatContext(): Promise<ChatContextResult | undefined> {
+        await this.ready;
+        return this.languageClient.sendRequest(CppContextRequest, null);
```

Comment:
Not sure if applicable, but `sendRequest` supports being passed a cancellation token.  Then, if the native side cancels, you'll get a different object back.  We've been detected native-side cancellation by checking if the resulting object lacks some otherwise required field.

Reply:
I plumbed the cancellation token through in the last push. I also added a helper to allow waiting for a promise with cancellation.